tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -106,18 +101,27 @@ spec:
           containerImage=$(jq -r '.containerImage' <<< $component)
           repository=$(jq -r '.repository' <<< $component)
           imageTags=$(jq '.tags' <<< $component)
-
           arches=$(get-image-architectures "${containerImage}" | jq -s 'map(.platform.architecture)')
+
+          # oras has very limited support for selecting the right auth entry,
+          # so create a custom auth file with just one entry.
+          AUTH_FILE=$(mktemp)
+          select-oci-auth "${containerImage}" > "$AUTH_FILE"
+
           # Just read the first from the list of architectures
-          arch=$(jq -r '.[0]' <<< $arches)
+          read -r arch_json <<< $(get-image-architectures "${containerImage}")
```

Comment:
Can you change this and line 113 to get the os like we get the arch? I think this is due to this change being started before I made the `arches` change. It should just be copy the `arches=` and `arch=` line but change `platform.os` instead of `platform.architecture` (and rename the variables of course)

Reply:
You could also just call `get-image-architectures` once and save it to a variable and then have a jq on it to set `arches` and a jq on it to set `oses`

Reply:
```
arch_json=$(get-image-architectures "${containerImage}")
arches=$(jq -s 'map(.platform.architecture) <<< $arch_json)
oses=$(jq -s 'map(.platform.os)' <<< $arch_json)
```